frontend/simple/js/event-log.js

```diff
@@ -0,0 +1,134 @@
+import localforage from 'localforage'
+import { toHash, makeEntry, makeGroup } from '../../../shared/functions'
+import request from 'superagent'
+
+// =======================
+// Helpers
+// =======================
+function getDB (hash) {
+  let db = localforage.createInstance({
+    name: 'Group Income',
+    storeName: hash
+  })
+  return db
+}
+function postEvent (groupId, hash, entry) {
+  return new Promise(function (resolve, reject) {
+    request.post(`${process.env.API_URL}/event/${groupId}`)
+      .send({hash, entry})
+      .end(function (err, res) {
+        if (err) {
+          return reject(err)
+        }
+        return resolve()
+      })
+  })
+}
+function postGroup (hash, entry) {
+  return new Promise(function (resolve, reject) {
+    request.post(`${process.env.API_URL}/group`)
+      .send({hash, entry})
+      .end(function (err, res) {
+        if (err) {
+          return reject(err)
+        }
+        return resolve(res.hash)
+      })
+  })
+}
+// =======================
+// Exports
+// =======================
+// get retrieves from the log
+// opts object allows for parent hash to be returned instead of data
+export async function get (log, opts) {
+  let hash
+  if (typeof opts === 'string') {
+    hash = opts
+  } else {
+    hash = opts.hash
+  }
+  let db = getDB(log.groupId)
```

Comment:
Because the instance is dependent upon the log that is passed in. The way the vuex mutates the log in order to put it into state object makes it hard to associate it with a specific instance of db without creating a new instance. There is not much going on in instance creation.